### ISSUE TITLE
tr/// modifies glob names?

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#1963]
Migrated from [rt.perl.org#3236](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3236) (status was 'resolved')

Searchable as RT3236$

**Comment by p5pRT** [comment]
### From @ysth

#### Created by @ysth



Ran across this while playing with a JAPH&#8203;:

\[D&#8203;:\\\]perl \-Wle"$&#8203;::\{STDOUT\}=~s/\\\*/ /;print"$&#8203;::\{STDOUT\}";print STDERR $&#8203;::\{STDOUT\}"
Modification of a read\-only value attempted at \-e line 1\.

Ok\, so far\, now lets try tr///&#8203;:

\[D&#8203;:\\\]perl \-Wle"$&#8203;::\{STDOUT\}=~tr/\*/ /;print"$&#8203;::\{STDOUT\}";print STDERR $&#8203;::\{STDOUT\}"
Filehandle  main&#8203;::STDOUT never opened at \-e line 1\.
\* main&#8203;::STDOUT

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.6.0:

Configured by sthoenna at Thu Mar 23 16:36:26 PST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=os2, osvers=2.30, archname=os2
    uname='os2 efn.org 2 2.30 i386  '
    config_args='-de -Dprefix=d:/perl -Aoptimize=-DDEBUGGING'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='gcc', optimize='-O2 -fomit-frame-pointer -malign-loops=2 -malign-jumps=2 -malign-functions=2 -s -DDEBUGGING', gccversion=2.8.1
    cppflags='-Zomf -Zmt -DDOSISH -DOS2=2 -DEMBED -I. -D_EMX_CRT_REV_=62'
    ccflags ='-Zomf -Zmt -DDOSISH -DOS2=2 -DEMBED -I. -D_EMX_CRT_REV_=62'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-Zexe -Zomf -Zmt -Zcrtdll -Zstack 32000'
    libpth=d:/emx/lib d:/emx/lib/mt
    libs=-lsocket -lm -lbsd
    libc=d:/emx/lib/mt/c_import.lib, so=dll, useshrplib=true, libperl=libperl.lib
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-Zdll', lddlflags='-Zdll -Zomf -Zmt -Zcrtdll -s'

Locally applied patches:
    


@INC for perl v5.6.0:
    d:/perl/lib/5.6.0/os2
    d:/perl/lib/5.6.0
    d:/perl/lib/site_perl/5.6.0/os2
    d:/perl/lib/site_perl/5.6.0
    d:/perl/lib/site_perl
    .


Environment for perl v5.6.0:
    HOME=d:\home\sthoenna
    LANG=en_us
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=d:\bin;C:\OS2;d:\perl\bin;C:\OS2\SYSTEM;C:\OS2\INSTALL;C:\;C:\OS2\MDOS;C:\OS2\APPS;C:\MMOS2;d:\os2apps\util;d:\DOSAPPS\UTIL;c:\sio;D:\WINDOWS;d:\pdksh;d:\emx\bin;d:\emacs\19.33\bin;d:\ispell
    PERL_BADLANG (unset)
    PERL_SH_DIR=d:\BIN
    SHELL (unset)


```
</details>

**Comment by p5pRT** [comment]
### From @gannett-ggreer

On Thu May 11 01&#8203;:34&#8203;:49 2000\, ysth wrote&#8203;:
> 
> Ran across this while playing with a JAPH&#8203;:
> 
> \[D&#8203;:\\\]perl \-Wle"$&#8203;::\{STDOUT\}=~s/\\\*/ /;print"$&#8203;::\{STDOUT\}";print STDERR
> $&#8203;::\{STDOUT\}"
> Modification of a read\-only value attempted at \-e line 1\.
> 
> Ok\, so far\, now lets try tr///&#8203;:
> 
> \[D&#8203;:\\\]perl \-Wle"$&#8203;::\{STDOUT\}=~tr/\*/ /;print"$&#8203;::\{STDOUT\}";print STDERR
> $&#8203;::\{STDOUT\}"
> Filehandle  main&#8203;::STDOUT never opened at \-e line 1\.
> \* main&#8203;::STDOUT

According to my tests\, this was fixed by Nicholas Clark in 1f257c956 on
December 27\, 2007\.  According to git that would be fixed before 5\.11\.0
so is that version tagged on this bug report accurate?

\-\- 
George Greer

**Comment by p5pRT** [comment]
### From @smpeters

On Fri Jul 16 20&#8203;:11&#8203;:51 2010\, greerga wrote&#8203;:
> On Thu May 11 01&#8203;:34&#8203;:49 2000\, ysth wrote&#8203;:
> > 
> > Ran across this while playing with a JAPH&#8203;:
> > 
> > \[D&#8203;:\\\]perl \-Wle"$&#8203;::\{STDOUT\}=~s/\\\*/ /;print"$&#8203;::\{STDOUT\}";print STDERR
> > $&#8203;::\{STDOUT\}"
> > Modification of a read\-only value attempted at \-e line 1\.
> > 
> > Ok\, so far\, now lets try tr///&#8203;:
> > 
> > \[D&#8203;:\\\]perl \-Wle"$&#8203;::\{STDOUT\}=~tr/\*/ /;print"$&#8203;::\{STDOUT\}";print STDERR
> > $&#8203;::\{STDOUT\}"
> > Filehandle  main&#8203;::STDOUT never opened at \-e line 1\.
> > \* main&#8203;::STDOUT
> 
> According to my tests\, this was fixed by Nicholas Clark in 1f257c956 on
> December 27\, 2007\.  According to git that would be fixed before 5\.11\.0
> so is that version tagged on this bug report accurate?


Yes\, that version would be correct\.  This ticket is now resolved\.

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
